engine/src/main/coffee/extensions/palette.coffee

```diff
@@ -0,0 +1,2340 @@
+# (C) Uri Wilensky. https://github.com/NetLogo/Tortoise
+
+{ exceptionFactory: exceptions } = require('util/exception')
+
+SingleObjectExtensionPorter = require('../engine/core/world/singleobjectextensionporter')
```

Comment:
I don't see the `SingleObjectExtensionPorter` used anywhere.  If it's not needed you can remove it.